stable/enterprise/templates/_common.tpl

```diff
@@ -247,6 +247,50 @@ app.kubernetes.io/component: {{ $component | lower }}
 {{- end -}}
 
 
+{{/*
+Create an image specification template that can override the default image
+based on global settings
+*/}}
+{{- define "enterprise.common.image" -}}
+{{- if eq (printf "%T" .Values.image) "string" }}
+  {{- .Values.image | trim -}}
+{{- else if and .Values.image.digest -}}
+  {{- printf "%s/%s@%s" .Values.image.registry .Values.image.repository .Values.image.digest | trim -}}
```

Comment:
in both these cases, tag or digest, if the user only fills out part of the dict, it will output a bad reference that will fail to run the pods. I'd like to see some checks that the registry and repository properties exist and are non-nil.

Reply:
Ack thanks @zhill, I'll revisit and get checks in place for this.

Reply:
Ok this should handle it better.